### ISSUE TITLE
»multiple rules generate« warning with seemingly no duplicate rules

### ISSUE THREAD

**Issue by Optiligence** [ninja-build/ninja#1298]
This dummy project creates a ninja build using CMake, _generating_ an input file for CMake’s AUTORCC. 

It works, but displays a »multiple rules generate« warning. 
The build.ninja file doesn’t seem to contain duplicate rules though. 
What’s going on here? 

[ninja dup test.zip](https://github.com/ninja-build/ninja/files/1120154/ninja.dup.test.zip)
<details> 
<summary>generated build.ninja for your convenience</summary>
<p>

```ninja
# CMAKE generated file: DO NOT EDIT!
# Generated by "Ninja" Generator, CMake Version 3.8

# This file contains all the build statements describing the
# compilation DAG.

# =============================================================================
# Write statements declared in CMakeLists.txt:
# 
# Which is the root file.
# =============================================================================

# =============================================================================
# Project: ninja-dup-test
# Configuration: 
# =============================================================================

#############################################
# Minimal version of Ninja required by this file

ninja_required_version = 1.5

# =============================================================================
# Include auxiliary files.


#############################################
# Include rules file.

include rules.ninja


#############################################
# Utility command for rebuild_cache

build CMakeFiles/rebuild_cache.util: CUSTOM_COMMAND
  COMMAND = cd "/home/user/ninja dup test/build" && /usr/bin/cmake -H"/home/user/ninja dup test" -B"/home/user/ninja dup test/build"
  DESC = Running CMake to regenerate build system...
  pool = console
  restat = 1
build rebuild_cache: phony CMakeFiles/rebuild_cache.util
# =============================================================================
# Object build statements for EXECUTABLE target ninja-dup-test


#############################################
# Order-only phony target for ninja-dup-test

build cmake_order_depends_target_ninja-dup-test: phony || ninja-dup-test_autogen generated.qrc
build CMakeFiles/ninja-dup-test.dir/main.cpp.o: CXX_COMPILER__ninja-dup-test ../main.cpp || cmake_order_depends_target_ninja-dup-test
  DEFINES = -DQT_CORE_LIB -DQT_NO_DEBUG
  DEP_FILE = CMakeFiles/ninja-dup-test.dir/main.cpp.o.d
  FLAGS = -fPIC -std=gnu++11
  INCLUDES = -isystem /usr/include/qt -isystem /usr/include/qt/QtCore -isystem /usr/lib/qt/mkspecs/linux-g++
  OBJECT_DIR = CMakeFiles/ninja-dup-test.dir
  OBJECT_FILE_DIR = CMakeFiles/ninja-dup-test.dir
build CMakeFiles/ninja-dup-test.dir/ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp.o: CXX_COMPILER__ninja-dup-test ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp || cmake_order_depends_target_ninja-dup-test
  DEFINES = -DQT_CORE_LIB -DQT_NO_DEBUG
  DEP_FILE = CMakeFiles/ninja-dup-test.dir/ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp.o.d
  FLAGS = -fPIC -std=gnu++11
  INCLUDES = -isystem /usr/include/qt -isystem /usr/include/qt/QtCore -isystem /usr/lib/qt/mkspecs/linux-g++
  OBJECT_DIR = CMakeFiles/ninja-dup-test.dir
  OBJECT_FILE_DIR = CMakeFiles/ninja-dup-test.dir/ninja-dup-test_autogen/JU62CA5L7X

# =============================================================================
# Link build statements for EXECUTABLE target ninja-dup-test


#############################################
# Link the executable ninja-dup-test

build ninja-dup-test: CXX_EXECUTABLE_LINKER__ninja-dup-test CMakeFiles/ninja-dup-test.dir/main.cpp.o CMakeFiles/ninja-dup-test.dir/ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp.o | /usr/lib/libQt5Core.so.5.9.1 || ninja-dup-test_autogen
  LINK_LIBRARIES = /usr/lib/libQt5Core.so.5.9.1
  OBJECT_DIR = CMakeFiles/ninja-dup-test.dir
  POST_BUILD = :
  PRE_LINK = :
  TARGET_FILE = ninja-dup-test
  TARGET_PDB = ninja-dup-test.dbg

#############################################
# Utility command for edit_cache

build CMakeFiles/edit_cache.util: CUSTOM_COMMAND
  COMMAND = cd "/home/user/ninja dup test/build" && /usr/bin/cmake-gui -H"/home/user/ninja dup test" -B"/home/user/ninja dup test/build"
  DESC = Running CMake cache editor...
  pool = console
  restat = 1
build edit_cache: phony CMakeFiles/edit_cache.util

#############################################
# Utility command for ninja-dup-test_autogen

build ninja-dup-test_autogen: phony CMakeFiles/ninja-dup-test_autogen ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp generated.qrc

#############################################
# Custom command for generated.qrc

build generated.qrc: CUSTOM_COMMAND
  COMMAND = cd "/home/user/ninja dup test/build" && /usr/bin/cmake -E copy_if_different /home/user/ninja\ dup\ test/source.qrc /home/user/ninja\ dup\ test/build/generated.qrc
  DESC = Generating generated.qrc
  restat = 1

#############################################
# Custom command for CMakeFiles/ninja-dup-test_autogen

build CMakeFiles/ninja-dup-test_autogen ninja-dup-test_autogen/JU62CA5L7X/qrc_generated.cpp: CUSTOM_COMMAND generated.qrc
  COMMAND = cd "/home/user/ninja dup test/build" && /usr/bin/cmake -E cmake_autogen "/home/user/ninja dup test/build/CMakeFiles/ninja-dup-test_autogen.dir/" ""
  DESC = Automatic RCC for target ninja-dup-test
  restat = 1
# =============================================================================
# Target aliases.

# =============================================================================
# Folder targets.

# =============================================================================
# =============================================================================
# Built-in targets


#############################################
# The main all target.

build all: phony ninja-dup-test

#############################################
# Make the all target the default.

default all

#############################################
# Re-run CMake if any of its inputs changed.

build build.ninja: RERUN_CMAKE | ../CMakeLists.txt /usr/lib64/cmake/Qt5/Qt5Config.cmake /usr/lib64/cmake/Qt5/Qt5ConfigVersion.cmake /usr/lib64/cmake/Qt5/Qt5ModuleLocation.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfig.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtras.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtrasMkspecDir.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigVersion.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreMacros.cmake /usr/share/cmake-3.8/Modules/AutogenInfo.cmake.in /usr/share/cmake-3.8/Modules/CMakeCInformation.cmake /usr/share/cmake-3.8/Modules/CMakeCXXInformation.cmake /usr/share/cmake-3.8/Modules/CMakeCommonLanguageInclude.cmake /usr/share/cmake-3.8/Modules/CMakeGenericSystem.cmake /usr/share/cmake-3.8/Modules/CMakeLanguageInformation.cmake /usr/share/cmake-3.8/Modules/CMakeParseArguments.cmake /usr/share/cmake-3.8/Modules/CMakeSystemSpecificInformation.cmake /usr/share/cmake-3.8/Modules/CMakeSystemSpecificInitialize.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU-C.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU-CXX.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU-C.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU-CXX.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU.cmake /usr/share/cmake-3.8/Modules/Platform/Linux.cmake /usr/share/cmake-3.8/Modules/Platform/UnixPaths.cmake CMakeCache.txt CMakeFiles/3.8.2/CMakeCCompiler.cmake CMakeFiles/3.8.2/CMakeCXXCompiler.cmake CMakeFiles/3.8.2/CMakeSystem.cmake generated.qrc
  pool = console

#############################################
# A missing CMake input file is not an error.

build ../CMakeLists.txt /usr/lib64/cmake/Qt5/Qt5Config.cmake /usr/lib64/cmake/Qt5/Qt5ConfigVersion.cmake /usr/lib64/cmake/Qt5/Qt5ModuleLocation.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfig.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtras.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigExtrasMkspecDir.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreConfigVersion.cmake /usr/lib64/cmake/Qt5Core/Qt5CoreMacros.cmake /usr/share/cmake-3.8/Modules/AutogenInfo.cmake.in /usr/share/cmake-3.8/Modules/CMakeCInformation.cmake /usr/share/cmake-3.8/Modules/CMakeCXXInformation.cmake /usr/share/cmake-3.8/Modules/CMakeCommonLanguageInclude.cmake /usr/share/cmake-3.8/Modules/CMakeGenericSystem.cmake /usr/share/cmake-3.8/Modules/CMakeLanguageInformation.cmake /usr/share/cmake-3.8/Modules/CMakeParseArguments.cmake /usr/share/cmake-3.8/Modules/CMakeSystemSpecificInformation.cmake /usr/share/cmake-3.8/Modules/CMakeSystemSpecificInitialize.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU-C.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU-CXX.cmake /usr/share/cmake-3.8/Modules/Compiler/GNU.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU-C.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU-CXX.cmake /usr/share/cmake-3.8/Modules/Platform/Linux-GNU.cmake /usr/share/cmake-3.8/Modules/Platform/Linux.cmake /usr/share/cmake-3.8/Modules/Platform/UnixPaths.cmake CMakeCache.txt CMakeFiles/3.8.2/CMakeCCompiler.cmake CMakeFiles/3.8.2/CMakeCXXCompiler.cmake CMakeFiles/3.8.2/CMakeSystem.cmake generated.qrc: phony

#############################################
# Clean all the built files.

build clean: CLEAN

#############################################
# Print all primary targets available.

build help: HELP
```
</p>
</details>

**Comment by bradking** [comment]
The "A missing CMake input file is not an error." section's phony rules incorrectly list `generated.qrc`, and there is a `build generated.qrc` build statement too.  This is a bug in CMake 3.8 that has been fixed in 3.9.  It is not a Ninja issue, so this can be closed IMO.

**Comment by Optiligence** [comment]
ok, thx
